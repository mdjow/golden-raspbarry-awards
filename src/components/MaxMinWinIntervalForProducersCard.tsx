"use client";

import { Card } from "./Card";
import { Table } from "./Table";
import { useProjections } from "../hooks/useProjections";
import { Producer } from "../types/Producer";

export function MaxMinWinIntervalForProducersCard() {
  const { max, min } = useProjections({
    projection: "max-min-win-interval-for-producers",
  });

  const columns = [
    {
      renderHeader: () => <span>Producer</span>,
      renderCell: (row: Producer) => <span>{row.producer}</span>,
    },
    {
      renderHeader: () => <span>Interval</span>,
      renderCell: (row: Producer) => <span>{row.interval}</span>,
    },
    {
      renderHeader: () => <span>Previous Year</span>,
      renderCell: (row: Producer) => <span>{row.previousWin}</span>,
    },
    {
      renderHeader: () => <span>following Year</span>,
      renderCell: (row: Producer) => <span>{row.followingWin}</span>,
    },
  ];

  return (
    <Card>
      <h3>Producers with longest and shortest interval between wins </h3>
      <p>Maximum </p>
      <Table columns={columns} data={max}></Table>
      <p>Minimum </p>
      <Table columns={columns} data={min}></Table>
    </Card>
  );
}
