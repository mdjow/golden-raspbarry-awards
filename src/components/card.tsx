"use client";

import styled from "styled-components";

export const Card = styled.div`
  display: flex;
  flex-direction: column;

  background: var(--bg-primary);
  backdrop-filter: blur(10px);
  border-radius: 0px 0px 4px 4px;
  box-shadow: 0 1px 3px 0 var(--shapes-dark);

  padding: 20px;
  min-height: 250px;
  min-width: 250px;

  h3 {
    font-weight: 600;
    font-size: 18px;
    line-height: 200%;
  }

  p {
    font-weight: 400;
    font-size: 18px;
    line-height: 150%;
  }
`;
