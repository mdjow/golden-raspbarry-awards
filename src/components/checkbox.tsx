import styled from "styled-components";

export const Checkbox = styled.input.attrs({ type: "checkbox" })`
  height: 15px;
  margin: 6px 0;
`;
