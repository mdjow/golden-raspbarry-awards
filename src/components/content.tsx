"use client";

import styled from "styled-components";

export const Content = styled.div`
  background-color: var(--main-background-color);
  display: flex;
  flex: 1;
`;
