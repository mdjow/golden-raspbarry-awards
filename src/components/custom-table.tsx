import React from "react";
import styled from "styled-components";

interface Column<T> {
  renderHeader: () => React.ReactNode;
  renderCell: (row: T) => React.ReactNode;
}

interface CustomTableProps<T> {
  columns: Column<T>[];
  data?: T[];
}

const Table = styled.table`
  width: 100%;
  border-collapse: collapse;
  margin-bottom: 20px;

  th,
  td {
    border: 1px solid #ccc;
    padding: 10px;
    text-align: left;
  }

  th {
    background-color: #f2f2f2;
    text-align: center;
  }

  th * {
    width: 100%;
  }
`;

export function CustomTable<T>({ columns, data }: CustomTableProps<T>) {
  if (!data?.length) {
    return <center>no results found</center>;
  }

  return (
    <Table>
      <thead>
        <tr>
          {columns.map((column, index) => (
            <th key={index}>{column.renderHeader()}</th>
          ))}
        </tr>
      </thead>
      <tbody>
        {data.map((row, rowIndex) => (
          <tr key={rowIndex}>
            {columns.map((column, colIndex) => (
              <td key={colIndex}>{column.renderCell(row)}</td>
            ))}
          </tr>
        ))}
      </tbody>
    </Table>
  );
}
