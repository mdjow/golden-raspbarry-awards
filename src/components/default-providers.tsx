"use client";

import { ReactNode } from "react";
import { QueryClientProvider } from "react-query";
import { ThemeProvider } from "styled-components";

import { queryClient } from "../services/queryClient";

interface DefaultProvidersProps {
  children: ReactNode;
}

const theme = {
  desktopBreakpoint: "968px",
  tableBreakpoint: "768px",
};

export function DefaultProviders({ children }: DefaultProvidersProps) {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </QueryClientProvider>
  );
}
