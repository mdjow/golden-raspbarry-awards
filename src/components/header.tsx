"use client";

import styled from "styled-components";
import { Saira_Stencil_One } from "next/font/google";

const sairaStencilOne = Saira_Stencil_One({
  weight: ["400"],
  subsets: ["latin"],
});

const MainHeader = styled.header`
  background-color: var(--header-background-color);
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 8px;
`;

const HeaderName = styled.a`
  color: var(--header-name-color);
  font-weight: 400;
  font-size: 26px;
  line-height: 150%;
`;

const Header: React.FC = () => {
  return (
    <MainHeader>
      <HeaderName className={sairaStencilOne.className}>
        Golden Raspberry Awards
      </HeaderName>
    </MainHeader>
  );
};

export default Header;
