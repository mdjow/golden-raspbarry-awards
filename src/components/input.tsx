import styled from "styled-components";

export const Input = styled.input<{ $inputColor?: string }>`
  padding: 6px;
  margin: 6px 0;
  color: ${(props) => props.$inputColor};
  border-color: var(--shapes-dark);
  border-radius: 3px;
`;
