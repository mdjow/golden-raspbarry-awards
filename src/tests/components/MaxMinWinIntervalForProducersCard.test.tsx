import renderer from "react-test-renderer";

import { MaxMinWinIntervalForProducersCard } from "../../components/MaxMinWinIntervalForProducersCard";
import { projectionsFetchResponseMock } from "./mocks/movieProjections.mock";

jest.mock("../../hooks/useProjections", () => ({
  useProjections: jest.fn(() => ({
    min: projectionsFetchResponseMock.min,
    max: projectionsFetchResponseMock.max,
  })),
}));

test("MaxMinWinIntervalForProducersCard component matches snapshot", () => {
  const component = renderer.create(<MaxMinWinIntervalForProducersCard />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
