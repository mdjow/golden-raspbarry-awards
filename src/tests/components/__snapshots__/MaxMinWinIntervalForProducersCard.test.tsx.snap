// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxMinWinIntervalForProducersCard component matches snapshot 1`] = `
<div
  className="sc-beyTiQ gWjloO"
>
  <h3>
    Producers with longest and shortest interval between wins 
  </h3>
  <p>
    Maximum 
  </p>
  <table
    className="sc-dmyDmy fSPGcv"
  >
    <thead>
      <tr>
        <th>
          <span>
            <span>
              Producer
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              Interval
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              Previous Year
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              following Year
            </span>
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span>
            Producer A
          </span>
        </td>
        <td>
          <span>
            3
          </span>
        </td>
        <td>
          <span>
            2010
          </span>
        </td>
        <td>
          <span>
            2013
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Producer B
          </span>
        </td>
        <td>
          <span>
            2
          </span>
        </td>
        <td>
          <span>
            2015
          </span>
        </td>
        <td>
          <span>
            2017
          </span>
        </td>
      </tr>
      <tr />
    </tbody>
  </table>
  <p>
    Minimum 
  </p>
  <table
    className="sc-dmyDmy fSPGcv"
  >
    <thead>
      <tr>
        <th>
          <span>
            <span>
              Producer
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              Interval
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              Previous Year
            </span>
          </span>
        </th>
        <th>
          <span>
            <span>
              following Year
            </span>
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span>
            Producer C
          </span>
        </td>
        <td>
          <span>
            1
          </span>
        </td>
        <td>
          <span>
            2012
          </span>
        </td>
        <td>
          <span>
            2013
          </span>
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Producer D
          </span>
        </td>
        <td>
          <span>
            1
          </span>
        </td>
        <td>
          <span>
            2018
          </span>
        </td>
        <td>
          <span>
            2019
          </span>
        </td>
      </tr>
      <tr />
    </tbody>
  </table>
</div>
`;
